datafusion/physical-plan/src/repartition/mod.rs

```diff
@@ -1042,18 +1042,6 @@ impl ExecutionPlan for RepartitionExec {
         &self,
         projection: &ProjectionExec,
     ) -> Result<Option<Arc<dyn ExecutionPlan>>> {
-        // If the projection does not narrow the schema, we should not try to push it down.
```

Comment:
I think this still shows up in the plans, in some cases the number of fields is bigger before repartitition because of this (and might hurt performance as `RepartitionExec` copies the input columns (currently twice)). Can we instead relax the conditition to `>` instead of `>=` so it will be pushed down if it also keeps the number of fields equal?